src/seismometer/data/pandas_helpers.py

```diff
@@ -97,10 +100,16 @@ def merge_windowed_event(
     event_time_col = event_time(event_label)
     event_val_col = event_value(event_label)
     one_event[r_ref] = one_event[event_time_col] - min_offset
-    predictions = _handle_merge(
-        predictions, one_event, pks, pred_ref=predtime_col, event_ref=r_ref, event_display=event_label
+
+    if merge_strategy=="count":
+        return _merge_event_counts(predictions, one_event, pks, event_val_col, window_hrs=window_hrs, min_offset=min_offset, l_ref=predtime_col, r_ref=r_ref)
```

Comment:
It could probably be done in a more consistent manner I agree, but I think that would require a refactor of merge_windowed_event. This refactor may get a tad messy to the point I'm not certain it would be cleaner or easier to read.

For _non-count merges_ the code will first merge the events based on the merge strategy. Once that's done it checks when those events happened inside of merge_windowed_event and if they happened outside of the merge window it replaces the event with pd.NA.
This isn't possible for _count merges_ because once the counting is done we no longer have access to when each event happened. Therefore we need to filter events outside the merge window before merging the counts. As a result we need a couple more args to be passed through when merging counts and we don't want to execute any of the additional code that happens after the merge normally. It seemed like the easiest way to accomplish this was to make a new function and just return the result of that. 

I'm open to refactoring though if you feel strongly. I can also just remove the windowing for counts-based merges but that feels inconsistent.

Reply:
Not a strong push.  I was actually expecting the "how/when to applying windowing is different", and wanted to validate my understanding.

Having the difference indicated somehow could be useful.

Reply:
Do you think a code comment would be sufficient for indicating the differences or would a refactor be preferred?

Reply:
Added a couple comments above the if statement and immediately inside it. Let me know if you'd like it to be more detailed

https://github.com/epic-open-source/seismometer/blob/4909413838d4a97c0ff81e3de0c5776d1d833642/src/seismometer/data/pandas_helpers.py#L115-L118